File: backend/app/schemas/wallet.py
from datetime import datetime

from pydantic import BaseModel, Field


class WalletResponse(BaseModel):
    id: int
    user_id: int
    balance: float
    currency: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class TopUpRequest(BaseModel):
    amount: float = Field(gt=0, le=10_000_000, description="Amount to top up in UZS")


class TopUpResponse(BaseModel):
    wallet: WalletResponse
    transaction_id: int
    reference_code: str
    message: str = "Wallet topped up successfully"

File: flutter_app/lib/features/wallet/domain/entities/wallet.dart
import 'package:equatable/equatable.dart';

class Wallet extends Equatable {
  final int id;
  final int userId;
  final double balance;
  final String currency;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Wallet({
    required this.id,
    required this.userId,
    required this.balance,
    required this.currency,
    required this.createdAt,
    required this.updatedAt,
  });

  String get formattedBalance =>
      '${balance.toStringAsFixed(0).replaceAllMapped(RegExp(r'(\d{1,3})(?=(\d{3})+(?!\d))'), (m) => '${m[1]},')} $currency';

  @override
  List<Object?> get props => [id, userId, balance, currency];
}
